lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,252 @@
+        end
+        [unreachable_ips.empty?, unreachable_ips]
+      end
+      raise StandardError, "couldn't connect to : #{unreachable_ips.join(',')}" unless success
+      true
+    end
+
+    def playbook_running?
+      pid = ar_method do
+        @object.customizations[:agent][:ansible_pid] if @object.customizations[:agent]
+      end
+      !pid.nil?
+    end
+
+    def run_playbook_command(ip, username, cmd)
+      result = {:finished => false}
+      @ssh ||= LinuxAdmin::SSH.new(ip, username, ssh_auth.auth_key)
+      ansible_process_pid = ansible_pid
+      if ansible_process_pid
+        process_running = @ssh.perform_commands(["sudo kill -0 #{ansible_process_pid}"])[:stdout]
+        unless process_running.empty?
+          result[:finished] = true
+          result[:stdout] = @ssh.perform_commands(["sudo cat /tmp/ansible.log"])[:stdout]
+          stop_agent
+        end
+      else
+        start_playbook_command(username, cmd)
+      end
+      result
+    end
+
+    def start_playbook_command(username, cmd)
+      pid, socket = create_agent(username)
+      @ssh.perform_commands(["sudo rm -f #{ANSIBLE_ERROR_LOG} #{ANSIBLE_LOG}"])
+      @ssh.perform_commands(["SSH_AUTH_SOCK=#{socket} nohup sudo -b -E #{cmd} 1> #{ANSIBLE_LOG} 2> #{ANSIBLE_ERROR_LOG} < /dev/null"])
+      ansible_process_pid = @ssh.perform_commands(["sudo pgrep -f 'sudo -b -E'"])[:stdout].split("\r").first
+      customize(:ansible_pid => ansible_process_pid, :agent_pid => pid, :agent_socket => socket)
+    end
+
+    def agent_exists?
+      agent_pid
+    end
+
+    def create_agent(username)
+      output = @ssh.perform_commands { |ssh| ssh.exec!("ssh-agent -a ssh_manageiq_#{id}") }
+      vars = {}
+      output.scan(/^(\w+)=([^;\n]+)/) { |k, v| vars[k] = v }
+      socket, pid = vars.values_at('SSH_AUTH_SOCK', 'SSH_AGENT_PID')
+      socket = "/#{username}/#{socket}"
+      socket = "/home#{socket}" if username != "root"
+      @ssh.perform_commands(["SSH_AUTH_SOCK=#{socket} SSH_AGENT_PID=#{pid} ssh-add -"], nil, ssh_auth.auth_key)
+      [pid, socket]
+    end
+
+    def stop_agent
+      pid =  agent_pid
+      socket = agent_socket
+      @ssh.perform_commands(["SSH_AGENT_PID=#{pid} ssh-agent -k &> /dev/null"])
+      @ssh.perform_commands(["sudo rm -f #{socket}"])
+      customize(:ansible_pid => nil, :agent_pid => nil, :agent_socket => nil)
+    end
+
+    def analyze_ansible_output(output)
+      if output.blank?
+        return false
+      end
+      results = output.rpartition('PLAY RECAP ********************************************************************').last
+      results = results.split("\r\n")
+      results.shift
+      results.detect { |x| !x.include?("unreachable=0") || !x.include?("failed=0") }.blank?
+    end
+
+    def agent_pid
+      ar_method do
+        @object.customizations[:agent][:agent_pid] if @object.customizations[:agent]
```

Comment:
You could use `@object.customizations.fetch_path(:agent, :agent_pid)` and get rid of the `if`